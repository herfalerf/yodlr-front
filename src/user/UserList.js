import React, { useEffect, useState } from "react";
import UserCard from "./UserCard";
import YodlrApi from "../api/api";

const UserList = () => {
  const [users, setUsers] = useState([]);

  useEffect(function getUsersOnMount() {
    console.debug("UserList useEffect getUsersOnMount");

    getUsers();
  }, []);

  async function getUsers() {
    let users = await YodlrApi.getUsersList();

    setUsers(users);
  }

  return (
    <div className="UserList col-md-8 offset-md-2">
      <div>
        {users.map((user) => (
          <UserCard
            id={user.id}
            firstName={user.firstName}
            lastName={user.lastName}
            email={user.email}
            state={user.state}
            key={user.id}
          />
        ))}
      </div>
    </div>
  );
};

export default UserList;
