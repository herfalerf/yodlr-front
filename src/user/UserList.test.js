import React from "react";
import { render } from "@testing-library/react";
import UserList from "./UserList";

it("renders without crashing", function () {
  render(<UserList />);
});
