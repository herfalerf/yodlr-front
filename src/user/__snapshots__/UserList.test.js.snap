// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no users 1`] = `
<DocumentFragment>
  <div
    class="UserList col-md-8 offset-md-2"
  >
    <div />
  </div>
</DocumentFragment>
`;
